expression/builtin_time_vec.go

```diff
@@ -1352,11 +1352,40 @@ func (b *builtinWeekWithoutModeSig) vecEvalInt(input *chunk.Chunk, result *chunk
 }
 
 func (b *builtinUnixTimestampDecSig) vectorized() bool {
-	return false
+	return true
 }
 
 func (b *builtinUnixTimestampDecSig) vecEvalDecimal(input *chunk.Chunk, result *chunk.Column) error {
-	return errors.Errorf("not implemented")
+	n := input.NumRows()
+
+	timeBuf, err := b.bufAllocator.get(types.ETTimestamp, n)
+	if err != nil {
+		return err
+	}
+	defer b.bufAllocator.put(timeBuf)
+	if err := b.args[0].VecEvalTime(b.ctx, input, timeBuf); err != nil {
+		return err
+	}
+	result.ResizeDecimal(n, false)
+	Ts := result.Decimals()
+	for i := 0; i < n; i++ {
+		if timeBuf.IsNull(i) {
```

Comment:
We can use `result.MergeNulls(timeBuf)` here.

Reply:
Thanks for your suggestion, I have modified